lib/saluki-core/src/data_model/event/log/mod.rs

```diff
@@ -0,0 +1,95 @@
+//! Logs.
+
+use std::num::NonZeroU64;
+
+use stringtheory::MetaString;
+
+/// Internal representation of a log event aligned to Datadog logs payload fields https://docs.datadoghq.com/api/latest/logs/#log-object.
+#[derive(Clone, Debug, PartialEq)]
+pub struct Log {
+    /// Log message body.
+    message: MetaString,
+    /// Log status/severity (e.g., "info", "warn", "error").
+    status: Option<MetaString>,
+    /// Timestamp in seconds since Unix epoch.
+    timestamp: Option<NonZeroU64>,
+    /// Hostname associated with the log.
+    hostname: MetaString,
+    /// Service associated with the log.
+    service: MetaString,
+    /// Source of the log (ddsource).
+    source: MetaString,
+    /// Comma-separated tags (ddtags), format: key:value,key:value.
+    tags: MetaString,
+}
+
+impl Log {
+    /// Creates a new `Log` with the given message.
+    pub fn new(message: impl Into<MetaString>) -> Self {
+        Self {
+            message: message.into(),
+            status: None,
+            timestamp: None,
+            hostname: MetaString::empty(),
+            service: MetaString::empty(),
+            source: MetaString::empty(),
+            tags: MetaString::empty(),
+        }
+    }
+
+    /// Sets the log status/severity.
+    pub fn with_status(mut self, status: impl Into<Option<MetaString>>) -> Self {
+        self.status = status.into();
+        self
+    }
+
+    /// Sets the timestamp in seconds since Unix epoch. `None` clears the timestamp.
```

Comment:
In Rust, doc comments have an implicit structure, which affects how they're used/displayed in IDE hover hints, in generated docs, and so on. The first line is meant to be a summary/high-level description, and any further detail should come as a new paragraph.

We already have an equivalent method for `EventD` which we should mirror here, in terms of its doc comment:

```rust
    /// Set the timestamp.
    ///
    /// Represented as a Unix timestamp, or the number of seconds since the Unix epoch.
```